### PR TITLE
Orca: update tf/spark runner return stats

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

https://github.com/intel-analytics/BigDL/issues/7257

The `history` of tf fit can be found at https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/History

The `tf.keras.callbacks.History` object contains the following useful information
+ history.params: a dict, e.g. `{verbose:1, epochs:2, steps:77}`
+ history.history: a dict, contains train stats, e.g. `{'loss': [0.4278483986854553, 0.33550089597702026], 'accuracy': ...}`

Currently, the returned history in `tf_runner` is: 
```python
stats = {"train_" + k: v[-1] for k, v in history.history.items()}
```

Improvements:
+ remove the unnecessary "train_" prefix (see below comments)
+ `v[-1]`  -> `history.history` return all the epochs instead of the last one only
+ Also return `history.params` as part of the dict 

### 2. Summary of the change 

Two files, the source file `tf_runner.py` and its corresponding testing py.
```python
assert stats["train_loss"] == 0.0
```
⬇️ 
```python
assert stats["loss"] == [0.0]
```
### 3. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

**Comment by hamham223** [comment]
Update:


The original return value of tf model fit does not have the prefix "train_". e.g. the following example:

```
{'loss': [0.4325478971004486, 0.331192284822464], 'accuracy': [0.8016195893287659, 0.8552353978157043], 'auc': [0.7759723663330078, 0.8810408115386963], 'precision': [0.5576444268226624, 0.7030420899391174], 'recall': [0.11731240898370743, 0.493164598941803], 'val_loss': [0.3604627847671509, 0.3141428530216217], 'val_accuracy': [0.8411620855331421, 0.8619629144668579], 'val_auc': [0.8537807464599609, 0.8923787474632263], 'val_precision': [0.6987895965576172, 0.7068519592285156], 'val_recall': [0.36547356843948364, 0.532505452632904]}
```
+ All the keys without `val_` means train result

How to test: using test data as validation data

```python
est.fit(train_data,
        validation_data=test_data,
        validation_steps=2,
        epochs=2,
        batch_size=batch_size,
        feature_cols=feature_cols,
        label_cols=label_cols,
        steps_per_epoch=train_steps,
        callbacks=callbacks,)
```

File: python/orca/src/bigdl/orca/learn/tf2/tf_runner.py [hunk 1]
```diff
@@ -409,7 +409,10 @@ def step(self, data_creator, epochs=1, batch_size=32, verbose=1,
         if history is None:
             stats = {}
         else:
-            stats = {"train_" + k: v[-1] for k, v in history.history.items()}
+            params = history.params
+            stats = {k: v[:] for k, v in history.history.items()}
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/tf_runner.py:413]: is v[:] necessary?
just params.update(history.history)?

File: python/orca/src/bigdl/orca/learn/tf2/spark_runner.py [hunk 2]
```diff
@@ -385,7 +385,14 @@ def step(self, data_creator, epochs=1, batch_size=32, verbose=1,
         if history is None:
             stats = {}
         else:
-            stats = {k: v[-1] for k, v in history.history.items()}
+            stats = dict()
+            # history is a callbacks object
+            # see https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/History
+            # history.parmas is a dict, e.g. {'verbose':1, 'epochs':2, 'steps':77}
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:391]: history.parmas is a dict including keys verbose, epochs and steps

File: python/orca/src/bigdl/orca/learn/tf2/spark_runner.py [hunk 3]
```diff
@@ -385,7 +385,14 @@ def step(self, data_creator, epochs=1, batch_size=32, verbose=1,
...
+            stats.update(history.params)
+            # history.history contains train stats
+            # e.g. {'loss': [0.4, 0.3], 'accuracy': [..., ...], ...}
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:394]: history.history contains train and val stats

File: python/orca/src/bigdl/orca/learn/tf2/spark_runner.py [hunk 4]
```diff
@@ -385,7 +385,14 @@ def step(self, data_creator, epochs=1, batch_size=32, verbose=1,
         if history is None:
             stats = {}
         else:
-            stats = {k: v[-1] for k, v in history.history.items()}
+            stats = dict()
+            # history is a callbacks object
+            # see https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/History
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/spark_runner.py:390]: remove these two lines in spark_runner, only keep in tf_runner

**Review by hkvision — APPROVED** [review]
